src/OpenTracing/ActiveSpanSource.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace OpenTracing;
+
+/**
+ * Keeps track of the current active `Span`.
+ */
+interface ActiveSpanSource
+{
+    const THREAD_DEFAULT = 'default';
+
+    /**
+     * Activates an `Span`, so that it is used as a parent when creating new spans.
+     * The implementation must keep track of the active spans sequence, so
+     * that previous spans can be resumed after a deactivation.
+     *
+     * @param Span $span
+     * @param string|array $threads Use only in case of async operations. Allows you to keep
+     * an span as active in different threads.
+     */
+    public function activate(Span $span, $threads = self::THREAD_DEFAULT);
```

Comment:
The use of $threads make no sense to me, because you assume that at every location in the code, it knows what its own thread id is. You cannot guarantee this.

In addition it puts the responsibility for the thread selection to the client, but the client doesn't necessary know what kind of threading context it is in. The ActiveSpanSource should now and the python implementation PR has examples how custom ActiveSpanSources handle this internally, not expose this to the client.

I am against this approach.